### PR TITLE
NO-TICKET - fix file downloads not fetching all chunks and undefined …

### PR DESCRIPTION
**Pull request by austin-rausch**
…labels

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -17,7 +17,8 @@
     "rootDir": "src",
     "strict": true,
     "strictNullChecks": false,
-    "target": "es2017",
+    "target": "es6",
```
Comment (austin-rausch) [inline tsconfig.json:20]: This transpiles for an earlier js version, not sure _exactly_ whats supported or intended to be supported by notebooks, but per [this page](https://www.typescriptlang.org/tsconfig#target) es6 seems like a good target.

File: package.json [hunk 2]
```diff
@@ -53,7 +53,7 @@
     "mkdirp": "^1.0.3",
     "prettier": "1.16.4",
     "rimraf": "^2.6.1",
-    "typescript": "~3.7.0",
+    "typescript": "^4.2.3",
```
Comment (austin-rausch) [inline package.json:56]: Build was failing, the installed version didn't have updated types for the readable streams, this version does.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        run: |
+          yarn
+          yarn lint
+          yarn test
+          yarn build
+      - name: Publish to npm
+        if: startsWith(github.ref, 'refs/tags/v')
+        run: |
+          yarn build
+          npm publish . --access restricted
```
Comment (austin-rausch) [inline .github/workflows/release.yml:32]: I _think_ this is what is intended to be published given the entries in `package.json` pointing to the `lib` directory.

Comment (RyanRatcliff) [reply .github/workflows/release.yml:32]: Not sure what you mean here about the restricted.  Mind telling me a bit more?  If we do restrict it, I'm not sure what, if at all, we'll have to update in the image build of jupyter to login etc.

Comment (austin-rausch) [reply .github/workflows/release.yml:32]: I just falsely assumed that this was a private package, not sure why I thought that. Switching it to public.

**Comment by austin-rausch** [comment]
Converting to draft while I play with the release action.

**Review by RyanRatcliff — APPROVED** [review]

**Review by ynotdraw — APPROVED** [review]

**Pull request merged**